common/txmgr/types/tx_store.go

```diff
@@ -76,8 +76,9 @@ type TransactionStore[
 	FindTxAttemptsRequiringResend(ctx context.Context, olderThan time.Time, maxInFlightTransactions uint32, chainID CHAIN_ID, address ADDR) (attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error)
 	// Search for Tx using the idempotencyKey and chainID
 	FindTxWithIdempotencyKey(ctx context.Context, idempotencyKey string, chainID CHAIN_ID) (tx *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error)
-	// Search for Tx using the fromAddress and sequence
-	FindTxWithSequence(ctx context.Context, fromAddress ADDR, seq SEQ) (etx *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error)
+	// Search for Tx for rebroadcast using the fromAddress and sequence
+	FindTxWithSequenceForRebroadcast(ctx context.Context, fromAddress ADDR, seq SEQ) (etx *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error)
```

Comment:
Renamed it, because `FindTxWithSequence` is inaccurate now, as we have a new state `finalized` which is ignored in current implementation, and we can't add it because the only by rebroadcast logic.

Therefore created another `FindFinalizedTxWithSequence` for the case which we want to fetch finalized state (only used by test now).

Let me know if you guys have a different opinion about this.